src/k8s-engine/lima.ts

```diff
@@ -330,6 +331,48 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
     return Promise.resolve(null);
   }
 
+  /**
+   * Check if the base (alpine) disk image is out of date; if yes, update it
+   * without removing existing data.  This is only ever called from updateConfig
+   * to ensure that the passed-in lima configuration is the one before we
+   * overwrote it.
+   */
+  protected async updateBaseDisk(currentConfig: LimaConfiguration) {
+    // Lima does not have natively have any support for this; we'll need to
+    // reach into the configuration and:
+    // 1) Figure out what the old base disk version is.
+    // 2) Confirm that it's out of date.
+    // 3) Change out the base disk as necessary.
+    // Unfortunately, we don't have a version string anywhere _in_ the image, so
+    // we will have to rely on the path in lima.yml instead.
+
+    const images = currentConfig.images.map(i => path.basename(i.location));
+    // We had a typo in the name of the image; it was "alpline" instead of "alpine".
+    const versionMatch = images.map(i => /^alpl?ine-lima-v([0-9.]+)-rd-/.exec(i)).find(defined);
+
+    if (!versionMatch) {
+      console.log(`Could not find base image version from ${ images }; skipping update of base images.`);
+
+      return;
+    }
+    const existingVersion = semver.coerce(versionMatch[1]);
+
+    if (!existingVersion || (semver.coerce(IMAGE_VERSION)?.compare(existingVersion) ?? 0) <= 0) {
```

Comment:
I assume the right hand side of the `||` expression is true if the bundled version is not newer than the already installed version. But it looks like RD will then just continue.

I think we can only continue if the versions are the same.

Otherwise we need a modal dialog that gives the user a choice between closing RD, or doing a "forced reset of the VM to an earlier version, losing all the workloads". We can't really support running a newer ISO with an older RD because we don't know if it will work, and it is not part of the test matrix.

Reply:
Expanded in 17ee47b463d7cd061e7129b62596d17c9ce84f84 to make the choices explicit.  Also changed it so it throws errors in the user's face if the existing disk is too new, or if it is an unexpected case.

The user can delete the VM manually after the error (since the startup process aborted).